src/cluster/placement/staged_placement.go

```diff
@@ -74,46 +71,42 @@ func newActiveStagedPlacement(
 	return p
 }
 
-func (p *activeStagedPlacement) ActivePlacement() (Placement, DoneFn, error) {
-	p.RLock()
-	placement, err := p.activePlacementWithLock(p.nowFn().UnixNano())
-	if err != nil {
-		p.RUnlock()
-		return nil, nil, err
-	}
-	return placement, p.doneFn, nil
-}
-
 func (p *activeStagedPlacement) Close() error {
-	p.Lock()
-	defer p.Unlock()
-
-	if p.closed {
+	if !p.closed.CAS(false, true) {
 		return errActiveStagedPlacementClosed
 	}
 	if p.onPlacementsRemovedFn != nil {
-		p.onPlacementsRemovedFn(p.placements)
+		pl, ok := p.placements.Load().(Placements)
```

Comment:
do you think it's worth adding a quick note that we understand that `pl` is still subject to mutability races, it's expected based on historical design, something to that effect?